include/srexception.h

```diff
@@ -101,7 +102,8 @@ class Exception: public std::exception
     Exception(const std::string& what_arg, const char* file, int line)
       : _msg(what_arg), _loc(file + std::string(":") + std::to_string(line))
     {
-        // NOP
+        Logger::get_instance().log_data(
+            LLInfo, exception_class() + " at " + _loc + ":" + _msg);
```

Comment:
Should we add a `log_exception` function to the logger to avoid exception classes having to re-implement this "{typename} at {loc}:{message}" format?

`log_exception` would add the benefit that if you choose to modify your logging format, all of the exception classes wouldn't have to update their implementation.

Reply:
We only have to do this in one place. All the other SmartRedis exception classes are getting their implementation through polymorphic inheritence